### PR TITLE
adjusted cmd/bisync/log.go indent function to be more expressive abou…

### PR DESCRIPTION
**Pull request by KDreynolds**
#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:34]: The `logf` part is great! I'm not sure we need the `msg` part. I think it's already clear we're in dry run mode from other lines, like `file.txt: Skipped copy as --dry-run is set`

It would also read a little awkwardly in some places, like:

```
NOTICE: - Path2    Do queued copies to in dry-run mode - Path1
```

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:33]: Very nice 🙂

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:39]: The idea here was to use different colors for Path1 and Path2, to make it easier to visually differentiate in the terminal. So I would suggest either removing this change or using a different color.

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:46]: This part needs another look, as it could be either direction here (1 to 2 or 2 to 1). So, as it stands now it would result in confusing lines like:

```
NOTICE: - Path2    Queue copy to Path2 Path1 in dry-run mode - file.txt
```

Here's an example of how this function is called:
https://github.com/rclone/rclone/blob/6e28edeb9aa1c18e27faf641729b2cbc259fe13a/cmd/bisync/deltas.go#L479

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:50]: I am confused about this... aren't both conditions doing the same thing? What was the intent of this change?

File: cmd/bisync/operations.go
Comment (nielash) [inline cmd/bisync/operations.go:227]: Note that this change would affect all runs, not just dry-runs. I don't necessarily mind it (I usually use `-v` anyway), but I wonder if some users would find it unnecessary?

**Review by nielash — CHANGES_REQUESTED** [review]
Thank you for this! I think it is very close 🙂 There are a few things that need another look -- I added some comments inline.

Also I note that some of the tests are failing, which is pretty common for anything that changes logging in Bisync -- it's an odd quirk of how the test suite is designed 😄 What you'll want to do is "goldenize" the tests -- the process is documented [here](https://rclone.org/bisync/#updating-golden-results). But I would save this as a last step, as some of them might end up resolving themselves as a result of the revisions.

Thanks again and let me know if you have questions!

File: cmd/bisync/log.go
Comment (KDreynolds) [reply cmd/bisync/log.go:50]: Sorry, this one is a mistake. My apologies.

File: cmd/bisync/log.go
Comment (KDreynolds) [reply cmd/bisync/log.go:39]: ill change to green, if that does not work, I can easily just remove.

File: cmd/bisync/log.go
Comment (KDreynolds) [reply cmd/bisync/log.go:34]: I'll remove the msg

File: cmd/bisync/operations.go
Comment (KDreynolds) [reply cmd/bisync/operations.go:227]: I'll change this back so as to not make out of scope changes.

**Comment by KDreynolds** [comment]
Ok I have made the suggested changes and attempted to address @nielash concerns. Please let me know if there is a better way or if this will work. Appreciate the chance to contribute!

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:38]: What this would do now is:

```
NOTICE: - Path1    Queue copy to Path2 Path2 - file.txt
```

So the destination path is getting duplicated.

I'm not sure that any change is necessary here, since the info is already in the `msg`.

File: cmd/bisync/log.go
Comment (nielash) [inline cmd/bisync/log.go:41]: We currently try to use green/yellow/red for their universal meanings of new/modified/deleted or success/warning/error, so I would choose a different color (or leave the default of `BlueFg`).

It should also avoid `CyanFg` (already used for Path1) and `MagentaFg` (already used as the default `msg` color). So that is why `BlueFg` was one of the only options left 🙂

**Review by nielash — COMMENTED** [review]
Thanks for the revision! Very close now 🙂 

In my opinion, the change from 32-35 is sufficient to fix the issue. So my advice would be: keep that change, remove the others, fix the tests, squash, and then we are good to go!

Also, you may want to [rebase on master](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#basing-your-changes-on-the-latest-master) before fixing the tests, as ncw recently made some changes to the tests in https://github.com/rclone/rclone/commit/f5f86786b27290964d3b6b29790f1d245f66aa1d

**Comment by KDreynolds** [comment]
ok I may have just totally confused myself with the indent function at this point. my apologies, I hope this addresses the issues how you were describing @nielash , if not, I'm ready to keep plugging away. Thank you for your patience on this.

**Review by nielash — COMMENTED** [review]
`log.go` is looking good to me now 👍

So I think all that's left now is a little bit of cleanup:

- "goldenize" to get the bisync tests to pass. This command will often get you most of the way there:

```
go test ./cmd/bisync -remote local -golden
```

You'll know when you got it right because all 10 [checks](https://github.com/rclone/rclone/pull/7647/checks) on this PR should turn green.

- let's try to cleanup the commit history a bit -- ideally it should be squashed into one commit containing only your changes (it looks like some unrelated changes got pulled in here too)

Thanks again for your help!

**Comment by KDreynolds** [comment]
ok so i had attempted to rebase with master and then squash into one commit. Then i 'goldenized' using "go test ./cmd/bisync -remote local -golden". I still had one test in bisync failing when i pushed. This is probably the issue that is causing the failing checks. I will investigate further. 

As for the unrelated changes i may have accidentally squashed into this commit, any guidance on fixing this mess i have made would be sincerely appreciated.

**Comment by nielash** [comment]
Nice job with the rebase! It looks to me like it worked to get rid of the unrelated changes 🙂 

I took a look at the test failures locally. For me, goldenizing fixed all but the `dry_run` test. On that one, I think the problem is that the regex change on this line:

https://github.com/rclone/rclone/blob/3ac06fbb708bd17a7f6ba2b4cc3a9969d9d4b504/cmd/bisync/bisync_test.go#L106

is no longer matching the log line it's supposed to find. I think that probably just reverting it should fix it, as I think we ended up reverting the change that required it in the first place.

I'm also now questioning if this should be reverted too:

https://github.com/rclone/rclone/blob/3ac06fbb708bd17a7f6ba2b4cc3a9969d9d4b504/cmd/bisync/bisync_test.go#L95-L97

as if anything the change would be from `INFO` to `NOTICE`, not the other way around... unfortunately it's a hard thing to test because the issue it fixes only happens on certain backends (not `local`), and whoever wrote that comment neglected to say which ones 😄 But I'm going to bet it's supposed to be `NOTICE`.

If there are any tests you can't get to pass, just get them as close as you can and then let me know and I can see if I can figure it out.

Also if you could squash your changes into one commit when you're done that would be ideal (although I think we can also take care of that during the merge, if you prefer!)

Thanks again!

**Review by nielash — APPROVED** [review]
I cleaned up the branch to fix the tests and they are passing now 🙂

This is now good to go, from my perspective! (Just pending final approval from @ncw)

I think the commits can be squashed during the merge process.

Thanks again for this contribution, @KDreynolds! I'm sure users will appreciate this fix 😃

**Comment by KDreynolds** [comment]
Thank you for all your help @nielash . I really appreciate it!

**Review by ncw — APPROVED** [review]
LGTM :-)

Please squash and merge!

**Pull request merged**
